src/ut/handlers_test.cpp

```diff
@@ -2916,56 +2977,9 @@ TEST_F(HandlersTest, IMSSubscriptionUserUnknownDeregCacheFail)
 
 TEST_F(HandlersTest, IMSSubscriptionOtherErrorCallReg)
 {
-  MockHttpStack::Request req(_httpstack,
-                             "/impu/" + IMPU + "/reg-data",
-                             "",
-                             "?private_id=" + IMPI,
-                             "{\"reqtype\": \"call\"}",
-                             htp_method_PUT);
-  req.method();
-  ImpuRegDataTask::Config cfg(true, 3600);
-  ImpuRegDataTask* task = new ImpuRegDataTask(req, &cfg, FAKE_TRAIL_ID);
-
-  MockCache::MockGetRegData mock_op;
-  EXPECT_CALL(*_cache, create_GetRegData(IMPU))
-    .WillOnce(Return(&mock_op));
-  EXPECT_DO_ASYNC(*_cache, mock_op);
-  task->run();
-
-  CassandraStore::Transaction* t = mock_op.get_trx();
-  ASSERT_FALSE(t == NULL);
-  EXPECT_CALL(mock_op, get_xml(_, _)).Times(AtLeast(1))
-    .WillRepeatedly(SetArgReferee<0>(IMPU_IMS_SUBSCRIPTION));
-  EXPECT_CALL(mock_op, get_registration_state(_, _)).Times(AtLeast(1))
-    .WillRepeatedly(SetArgReferee<0>(RegistrationState::NOT_REGISTERED));
-  EXPECT_CALL(mock_op, get_associated_impis(_)).Times(AtLeast(1))
-    .WillRepeatedly(SetArgReferee<0>(IMPI_IN_VECTOR));
-  EXPECT_CALL(mock_op, get_charging_addrs(_)).Times(AtLeast(1))
-    .WillRepeatedly(SetArgReferee<0>(NO_CHARGING_ADDRESSES));
-
-  EXPECT_CALL(*_mock_stack, send(_, _, 200))
-    .Times(1)
-    .WillOnce(WithArgs<0,1>(Invoke(store_msg_tsx)));
-  std::string error_text = "error";
-  t->on_success(&mock_op);
-
-  ASSERT_FALSE(_caught_diam_tsx == NULL);
-  Diameter::Message msg(_cx_dict, _caught_fd_msg, _mock_stack);
-  Cx::ServerAssignmentRequest sar(msg);
-
-  Cx::ServerAssignmentAnswer saa(_cx_dict,
-                                 _mock_stack,
-                                 0,
-                                 0,
-                                 0,
-                                 "",
-                                 NO_CHARGING_ADDRESSES);
-
-  EXPECT_CALL(*_httpstack, send_reply(_, 500, _));
-  _caught_diam_tsx->on_response(saa);
-
-  delete _caught_diam_tsx; _caught_diam_tsx = NULL;
-  _caught_fd_msg = NULL;
+  sa_error_template(0, 0, 500);
+  sa_error_template(DIAMETER_AUTHORIZATION_REJECTED, 0, 500);
+  sa_error_template(0, DIAMETER_ERROR_ROAMING_NOT_ALLOWED, 500);
 }
```

Comment:
Thanks.  Actually, now that I've reverted changes to handlers.cpp (apart from comments underlining the fact that there will only be at most one of result code or experimental code at any given time) I don't need these test changes, so I'll back these out as well